### PR TITLE
Remove OAuth2 authentication backend.

### PR DESCRIPTION
**Pull request by rocha**
OpenID Connect is now the primary authentication and authorization
backend.

### CODE REVIEW

**Comment by rocha** [comment]
@clintonb @dsjen Removing code, my favorite thing to do!

File: analytics_dashboard/analytics_dashboard/backends.py [hunk 1]
```diff
@@ -75,3 +37,12 @@ def get_user_details(self, response):
             # Optional, scope-specific, data
             u'language': response.get('language')
         }
+
+    def get_user_permissions(self, access_token):
+        # TODO: Do we need to worry about refreshing the token?
```
Comment (clintonb) [inline analytics_dashboard/analytics_dashboard/backends.py:41]: One day we'll actually answer this question. :P.

**Comment by clintonb** [comment]
:+1:

File: analytics_dashboard/analytics_dashboard/backends.py (hunk 1 above)
Comment (rocha) [inline analytics_dashboard/analytics_dashboard/backends.py:41]: Oh, it is on my TODO of "tweaks". :grin:

**Comment by clintonb** [comment]
Looks like you have a `make quality` error.

File: analytics_dashboard/analytics_dashboard/tests/test_views.py [hunk 2]
```diff
@@ -188,25 +191,53 @@ def _access_token_body(self, request, _url, headers, username):
...
-        token_response = {
+        client_secret = settings.SOCIAL_AUTH_EDX_OIDC_SECRET
+        access_token = {
             'access_token': '12345',
             'refresh_token': 'abcde',
             'expires_in': 900,
             'preferred_username': username
         }
-        return token_response
+        access_token.update({
```
Comment (dsjen) [inline analytics_dashboard/analytics_dashboard/tests/test_views.py]: Instead of `update`, would it make sense to combine this with the above declaration?

Comment (rocha) [inline analytics_dashboard/analytics_dashboard/tests/test_views.py]: Totally. I just did dumb copy pasta.

Comment (dsjen) [inline analytics_dashboard/analytics_dashboard/tests/test_views.py]: Mmm, pasta! :spaghetti:

**Comment by rocha** [comment]
@dsjen I addressed your comment.

**Comment by dsjen** [comment]
Looks good to me!

**Pull request merged**